### DISCUSSION TITLE
Is there any way to support read-replica pooling for specific queries?

### DISCUSSION THREAD

**Discussion by steadweb** [sidorares/node-mysql2#1451] [General]
Could you recommend an approach (if possible) to setup configuration supporting read-replica based pools. Ideally, this would select a read-replica connection from a specific pool is the query being run was a SELECT only, and didn't run INSERT, UPDATE or DELETE (as an example).

Sequelize has an example here: https://sequelize.org/master/manual/read-replication.html

**Comment by sidorares** [comment]
I don't think it's possible at this level. Sequelize knows when it generates reads and writes and can direct to an appropriate connection. This driver does not attempt to parse your sql query and classify it as read or write _before_ its sent to a server
